reactor-rxjava-guide.md

```diff
@@ -0,0 +1,119 @@
+
+## Background
+
+[Reactive Streams](http://www.reactive-streams.org/) is an industry standard for declarative dataflow programming in an asynchronous environment. More detail on design principles can be found in the [Reactive Manifesto](https://www.reactivemanifesto.org/). It is the basis for Azure's async Java SDKs going forward.
+
+A Reactive Streams framework implements the Reactive Streams Standard for specific programming languages. 
+
+The [RxJava](https://github.com/ReactiveX/RxJava) ([ReactiveX](http://reactivex.io/) for JVM) framework was the basis of past Azure Java SDKs, but will not be going forward. Async Java SDK 2.x.x was implemented using RxJava 1; in this guide we will assume that RxJava 1 is the version you are already familiar with i.e. as a result of working with the Async Java SDK 2.x.x.
+
+[Project Reactor](https://projectreactor.io/) or just *Reactor* is the Reactive Programming framework being used for new Azure Java SDKs. The purpose of the rest of this document is to help you get started with Reactor.
+
+## Comparison between Reactor and RxJava
+
+RxJava 1 provides a framework for implementing the **Observer Pattern** in your application. In the Observer Pattern,
+* ```Observable```s are entities that receive events and data (i.e. UI, keyboard, TCP, ...) from outside sources, and make those events and data available to your program.
+* ```Observer```s are the entities which subscribe to the Observable events and data.
+
+The [Reactor pattern guide](reactor-pattern-guide.md) gives a brief conceptual overview of Reactor. In summary:
+* ```Publisher```s are the entities which make events and data from outside sources available to the program
+* ```Subscriber```s subscribe to the events and data from the ```Publisher```
+
+Both frameworks facilitate asynchronous, event-driven programming. Both frameworks allow you to chain together a pipeline of operations between Observable/Observer or Publisher/Subscriber.
+
+Roughly, what you would use an ```Observable``` for in RxJava, you would use a ```Flux``` for in Reactor. And what you would use a ```Single``` for in RxJava, you would use a ```Mono``` for in Reactor.
+
+The critical difference between the two frameworks is really in the core implementation: 
+Reactor operates a service which receives event/data pairs serially from a ```Publisher```, demultiplexes them, and forwards them to registered ```Subscribers```. This model was designed to help servers efficiently dispatch requests in a distributed system.
+The RxJava approach is more general-purpose. ```Observer```s subscribe directly to the ```Observable``` and the ```Observable``` sends events and data directly to ```Observer```s, with no central service handling dispatch. 
```

Comment:
The main difference is reactorstream is for java8+.

but rxjava works with java7 and earlier as well. reactorstream takes advantage of java8+. and the guideline is any application on java8+ should use reactorstream